### PR TITLE
fix: use original modifiedIndex if possible

### LINKED ISSUE #4160: bug: Seek help !Problems of Prometheus, This should be a bug, please fix it!!!
### Environment

* apisix 2.4
* centos 7.6
* nginx version: openresty/1.19.3.1
* etcdctl version: 3.4.13
* apisix-dashboard 2.5

My APISIX 2.4 Prometheus plug-in reported "500 Internal Server Error" after running for a period of time. After restarting apisix, it returned to normal, and a 500 error was reported after a while

1.error.log
`2021/04/29 18:11:58 [warn] 17627#17627: *17251 [lua] ctx.lua:153: __index(): failed to fetch cookie value by key: cookie_ymmoa_passport error: no cookie foun
d in the current request, client: 10.13.65.71, server: , request: "POST /inner/loaner/query-user-auth-info HTTP/1.1", host: "loaner.dev-ag.56qq.com"
2021/04/29 18:11:58 [warn] 17627#17627: *17251 [lua] ctx.lua:153: __index(): failed to fetch cookie value by key: cookie_dev_passport error: no cookie found 
in the current request, client: 10.13.65.71, server: , request: "POST /inner/loaner/query-user-auth-info HTTP/1.1", host: "loaner.dev-ag.56qq.com"
2021/04/29 18:12:00 [error] 17627#17627: *18598 lua entry thread aborted: runtime error: ...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:206: att
empt to compare number with string
stack traceback:
coroutine 0:
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua: in function 'set_modify_index'
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:229: in function 'etcd_modify_index'
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:295: in function 'handler'
	/home/poseidon/apisix-2.4/apisix/api_router.lua:130: in function 'handler'
	...eidon/apisix-2.4//deps/share/lua/5.1/resty/radixtree.lua:720: in function 'dispatch'
	/home/poseidon/apisix-2.4/apisix/api_router.lua:165: in function 'match'
	/home/poseidon/apisix-2.4/apisix/init.lua:292: in function 'http_access_phase'
	access_by_lua(nginx.conf:205):2: in main chunk, client: 10.10.10.100, server: , request: "GET /apisix/prometheus/metrics HTTP/1.1", host: "10.10.10.100:80"
2021/04/29 18:12:15 [error] 17627#17627: *19213 lua entry thread aborted: runtime error: ...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:206: att
empt to compare number with string
stack traceback:
coroutine 0:
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua: in function 'set_modify_index'
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:229: in function 'etcd_modify_index'
	...seidon/apisix-2.4/apisix/plugins/prometheus/exporter.lua:295: in function 'handler'
	/home/poseidon/apisix-2.4/apisix/api_router.lua:130: in function 'handler'
	...eidon/apisix-2.4//deps/share/lua/5.1/resty/radixtree.lua:720: in function 'dispatch'
	/home/poseidon/apisix-2.4/apisix/api_router.lua:165: in function 'match'
	/home/poseidon/apisix-2.4/apisix/init.lua:292: in function 'http_access_phase'
	access_by_lua(nginx.conf:205):2: in main chunk, client: 10.10.10.100, server: , request: "GET /apisix/prometheus/metrics HTTP/1.1", host: "10.10.10.100:80"
`
2.my config.yaml:
etcd:
    host:
      - "http://10.10.0.10:2379"
apisix:
  admin_key:
    - name: "admin"
      # yamllint disable rule:comments-indentation
      key: 123456789 # using fixed API token has security risk, please update it when you deploy to production environment
      # yamllint enable rule:comments-indentation
      role: admin
plugins:
  - prometheus
  - log-rotate
  - proxy-rewrite
  - redirect
plugin_attr:
  prometheus:
    export_uri: /apisix/prometheus/metrics
    server-info:
      report_interval: 60
      report_ttl: 3600
  log-rotate:
    interval: 86400
    max_kept: 30
nginx_config:
  http:
    access_log_format: "$time_iso8601|$msec|$status|$request_completion|$bytes_sent|$body_bytes_sent|$realip_remote_addr|$remote_addr|$http_x_forwarded_for|$
remote_user|$host|$server_name|$server_port|$server_protocol|$scheme|$request_method|$request_length|$request_time|$request_uri|$uri|$content_length|$content_type|$http_referer|$http_user_agent|$http_app_jb|$http_client_info|$upstream_addr|$upstream_connect_time|$upstream_header_time|$upstream_response_time|$upstream_status|$upstream_bytes_received|$upstream_cache_status|$upstream_http_content_type|$upstream_http_content_length|$upstream_http_content_disposition|$http_x_cat_parent_id|$upstream_scheme://$upstream_host$upstream_uri"

Seek help !How to solve this problem?

### PR DESCRIPTION
**Pull request by spacewander**
Fix #4160

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
I test dozens of times with the reproduce steps from @tzssangglass and also this new test on master branch, and both of them passed without errors. Is there any specific reason to cause this problem, since it seems not always happen.

And if it is triggered randomly, do we need to test it several times to ensure the error would not occur

File: t/plugin/prometheus3.t [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugin_config_id": 1,
+                    "plugins": {
```
Comment (Yiyiyimu) [inline t/plugin/prometheus3.t:61]: maybe could remove `plugins` here

**Comment by spacewander** [comment]
@Yiyiyimu 
It can be simply reproduced with running my test without my fix.

**Comment by Yiyiyimu** [comment]
> @Yiyiyimu
> It can be simply reproduced with running my test without my fix.

Could reproduce for now. Can't find the reason yet maybe there is something configured wrong on my local environment..

**Review by Yiyiyimu — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**